README.md

```diff
@@ -1,103 +1,441 @@
+    ```shell
+    cp .env.development.local.example .env.development.local
+    ```
+
+    The most important things to set a local API are:
+    1. set Kultus API endpoints
+      ```shell
+      VITE_APP_API_URI=http://localhost:8081/graphql
+      VITE_APP_API_REPORT_URI=https://localhost:8081/reports
+      ```
+    2. set LinkedEvents endpoint
+      ```shell
+      VITE_APP_LINKEDEVENTS_API_URI=http://localhost:8080/v1
+      ```
+    3. Set OIDC client. Remember to set also the API to match these values.
+      ```shell
+      # authorization server sets in JWT as a authorized party and audience.
+      VITE_APP_OIDC_AUDIENCES="kultus-api-dev,profile-api-test"
+      # API client id as in audience
+      VITE_APP_OIDC_API_CLIENT_ID=kultus-api-dev
+      # authorized party
+      VITE_APP_OIDC_CLIENT_ID=kultus-admin-ui-dev
+      ```
+
+    > NOTE: if you have any problems to connect, double check the network port where the API is running. Running Docker services and their port forwardings can be checked with `docker ps -a`. 
+    >
+    > If you connect to a Docker container from your local computer, you can usually use https://localhost:8081/ to reach the API. 
+    
+5. Run the Node application in development mode with `yarn dev`. At this point you should be also running the local APIs if you have selected to use them.
+
+
+##### Running using Docker
+
+Using the following instructions you should be able to:
+
+- Run this UI using Docker & Docker compose
+- Run the Kultus backend locally in Docker or use the public test environment's backend
+
+1. Copy [.env.example](./.env.example) file to ./.env file).
+    ```shell
+    cp .env.example .env
+    ```
+2. Configure API endpoints
+    
+    **Kultus API**
+    ```shell
+    VITE_APP_API_URI=https://kultus.api.test.hel.ninja/graphql
+    VITE_APP_API_REPORT_URI=https://kultus.api.test.hel.ninja/reports
+    ```
+
+    **LinkedEvents**
+    ```shell
+    VITE_APP_LINKEDEVENTS_API_URI=https://linkedevents.api.test.hel.ninja/v1
+    ```
+
+    to use the API from the public test environment.
+    > NOTE: If you connect to a Docker container from your local computer, you can usually use https://localhost:8081/ to reach the API, but if you connect from a Docker container to another, you should be using the container names like http://kultus-backend:8081/ to reach the API or use http://host.docker.internal:8081/ to connect via your local machine ports.
+
+
+3. To build and start a docker container, you can use
+    **a development build:**
+    > NOTE: [compose.yml](./compose.yml) maps some local directories as a volume in order to get a hot reload for the code to work.
+    ```shell
+    DOCKER_TARGET=development docker compose up --build
+    ```
+
+    or 
+
+    if you want to run **a production build**, you most likely first need to unmap the composer volumes from [compose.yml](./compose.yml), by removing or fully commenting out the following section:
```

Comment:
Another way to make the development and production builds both work with Docker compose locally would be to use compose.yml and compose.override.yml files to achieve that (Haven't thought out the exact details in my head, but I think it'd probably be doable that way). The upside would be that user wouldn't have to manually change compose.yml which is committed to the repository and will therefore show up as changes. Not something necessary to do in this PR though.

Reply:
I think that using override-file is not a good practise in this kind of cases. I think that it should be left for those rare cases when you really need to override something in your local setup In cases when you need important local changes, but don't want to change any files in repository. I would prefer using [profiles](https://docs.docker.com/compose/how-tos/profiles/) or another named configuration file (e.g compose-production-build.yaml) instead. But, we do already have a DOCKER_TARGET variable for almost the same purpose.

Reply:
I'm also thinking, that we should not provide a docker build stage for development purposes at all. We _don't have to_ support that kind of special cases, eventhough we can do that. We don't need to provide a working docker environment for development use and we can never provide all the tools anyway. Tools like editors and such with all their settings will always be something personal. It's easier, faster, simpler and convenient way to just install a node in your local computer, when you develop a node application. Whether somebody wants to do that in a docker container, he can do that, but we don't need to provide support for that.

Reply:
So, I would be happy to remove the develop stage from dockerfile, remove the volumes mapping totally and make the production build stage the default stage.

Reply:
> So, I would be happy to remove the develop stage from dockerfile, remove the volumes mapping totally and make the production build stage the default stage.

Please, let's discuss this live.